### PR TITLE
codeintel: Cleanup uploads package

### PR DESCRIPTION
**Pull request by efritz**
This PR breaks the dependency from internal/background -> svc and instead uses its direct dependencies instead.

## Test plan

Existing unit tests.

### CODE REVIEW

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.00% (0.00 kb) | 0.00% (0.00 kb) | 0.00% (0.00 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/ef/cleanup-uploads-bkg/compare-report.html;tab=live_object) for a full comparison between the commits 736cde4 and 5a3ba25 or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

File: enterprise/internal/codeintel/uploads/internal/background/iface.go [hunk 1]
```diff
@@ -76,3 +39,19 @@ type GitserverClient interface {
...
 	RequestRepoUpdate(context.Context, api.RepoName, time.Duration) (*protocol.RepoUpdateResponse, error)
 }
+
+type Locker interface {
+	Lock(ctx context.Context, key int32, blocking bool) (bool, locker.UnlockFunc, error)
+}
+
+type PolicyService interface {
+	GetConfigurationPolicies(ctx context.Context, opts policiesshared.GetConfigurationPoliciesOptions) ([]codeinteltypes.ConfigurationPolicy, int, error)
+}
```
Comment (Strum355) [inline enterprise/internal/codeintel/uploads/internal/background/iface.go:49]:
```suggestion
}

```

File: enterprise/internal/codeintel/uploads/internal/background/job_commitgraph.go [hunk 2]
```diff
@@ -4,11 +4,139 @@ import (
...
+// lockAndUpdateUploadsVisibleToCommits will call UpdateUploadsVisibleToCommits while holding an advisory lock to give exclusive access to the
+// update procedure for this repository. If the lock is already held, this method will simply do nothing.
+func (s *commitGraphUpdater) lockAndUpdateUploadsVisibleToCommits(ctx context.Context, repositoryID, dirtyToken int, maxAgeForNonStaleBranches time.Duration, maxAgeForNonStaleTags time.Duration) (err error) {
+	// ctx, trace, endObservation := s.operations.updateUploadsVisibleToCommits.With(ctx, &err, observation.Args{
+	// 	LogFields: []log.Field{
+	// 		log.Int("repositoryID", repositoryID),
+	// 		log.Int("dirtyToken", dirtyToken),
+	// 	},
+	// })
+	// defer endObservation(1, observation.Args{})
```
Comment (Strum355) [inline enterprise/internal/codeintel/uploads/internal/background/job_commitgraph.go:66]: TODO?

Comment (efritz) [reply enterprise/internal/codeintel/uploads/internal/background/job_commitgraph.go:66]: Might not wrap this method in observability operation and might just emit explicit stats instead in most background jobs.

**Review by Numbers88s — APPROVED** [review]

**Pull request merged**